csrc/scheduler/matmul_utils.cpp

```diff
@@ -353,6 +478,24 @@ std::string getMatmulCompileTimeRejectReason(Fusion* fusion) {
   return "";
 }
 
+bool isCpAsyncOperandLoadSupported(
+    const MatmulParams* params,
+    int64_t dtype_size_a,
+    int64_t dtype_size_b) {
+  if (!isAmpere(params->mma_macro)) {
+    return false;
+  }
+  // Use cp.async for loading operands if vec size is compatible
+  const auto& validCpAsyncVecSize = [](int64_t dtype_size,
+                                       int64_t vec_size) -> bool {
+    int64_t cp_bytes = dtype_size * vec_size;
+    return cp_bytes == 16 || cp_bytes == 8 || cp_bytes == 4;
+  };
+  return params->double_buffer_options.smem_double_buffer_stage > 1 &&
+      validCpAsyncVecSize(dtype_size_a, params->supported_vec_size.a) &&
+      validCpAsyncVecSize(dtype_size_b, params->supported_vec_size.b);
```

Comment:
I think so. In most cases these will be the same but I don't think there's any harm in making them independent.

Reply:
Would it make sense to have a different number of load stages for the two operands? If A cannot be asynchronously loaded due to too small vec size then it might only be able to double buffer, while B could still be circular buffered. I will try this out to see if the resulting kernel runs.

Reply:
This change could potentially get large so I am thinking I will wait to do that in a follow-up PR.  See #2211 for tracking issue.

Reply:
I don't think this will work. IIRC, currently our double buffering lowering pass assumes all double/circular buffers at the same level has the same stage.